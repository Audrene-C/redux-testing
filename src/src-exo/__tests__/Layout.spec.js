import { shallow, mount } from 'enzyme';
import Home from '../components/Home';
import Layout from '../components/Layout';
import '../setUpTests';
import renderer from 'react-test-renderer';

//Créer un mock de Home
jest.mock("../components/Home", () => {
    return "home-component";
});

describe('Layout snapshot',()=>{

    //check if Layout renders without crashing
    it('It renders without crashing', () => {
        //Créer le expect
        let wrapper = shallow(<Layout />);
        expect(wrapper);
    });

    it('It renders correctly', () => {
        //Créer snapshot
        let wrapper = shallow(<Layout />);
        expect(wrapper).toMatchSnapshot();
    })

    it('It renders Home', () => {
        const wrapper = mount(<Layout />);
        let home = wrapper.find('home-component');
        expect(home.props.fetchRandomNumber);
    });
});
