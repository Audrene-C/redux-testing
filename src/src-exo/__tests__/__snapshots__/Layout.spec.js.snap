// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout snapshot It renders correctly 1`] = `ShallowWrapper {}`;
